src/vm/runners/cairo_runner.rs

```diff
@@ -426,6 +427,41 @@ impl CairoRunner {
         vm.segments.get_memory_holes(&builtin_accessed_addresses)
     }
 
+    /// Check if there are enough trace cells to fill the entire diluted checks.
+    pub fn check_diluted_check_usage(
+        &self,
+        vm: &VirtualMachine,
+    ) -> Result<(), VirtualMachineError> {
+        let diluted_pool_instance = match &self.layout.diluted_pool_instance_def {
+            Some(x) => x,
+            None => return Ok(()),
+        };
+
+        let mut used_units_by_builtins = 0;
+        for (_, builtin_runner) in &vm.builtin_runners {
+            let used_units = builtin_runner.get_used_diluted_check_units(
+                diluted_pool_instance.spacing,
+                diluted_pool_instance.n_bits,
+            );
+
+            used_units_by_builtins += used_units
+                * safe_div_usize(
```

Comment:
you can make this more readable creating a variable to hold the result of the safe_div